src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -690,6 +703,87 @@ private static bool IsNodeOrAnyAncestorLeftSideOfDot(SyntaxNode node, ISyntaxFac
 
             return false;
         }
+
+        internal static ImmutableArray<FindUsageProperty> GetAdditionalFindUsagesProperties(SyntaxNode node, SemanticModel semanticModel, ISyntaxFactsService syntaxFacts)
+        {
+            var additionalProperties = new ArrayBuilder<FindUsageProperty>();
+
+            if (GetInfo(
+                syntaxFacts.GetContainingTypeDeclaration(node, node.SpanStart),
+                ContainingTypeInfoPropertyName,
+                semanticModel,
+                out var findUsagePropertyForContainingType))
+            {
+                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingType);
+            }
+
+            if (GetInfo(syntaxFacts.GetContainingMemberDeclaration(node, node.SpanStart),
+                ContainingMemberInfoPropertyName,
+                semanticModel,
+                out var findUsagePropertyForContainingMember))
+            {
+                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingMember);
+            }
+
+            return additionalProperties.ToImmutableAndFree();
+        }
+
+        internal static ImmutableArray<FindUsageProperty> GetAdditionalFindUsagesProperties(ISymbol definition)
+        {
+            var additionalProperties = new ArrayBuilder<FindUsageProperty>();
+
+            var containingType = definition.ContainingType;
+            if (containingType != null && 
+                TryGetAdditionalProperty(ContainingTypeInfoPropertyName, containingType, out var findUsagePropertyForContainingType))
+            {
+                additionalProperties.Add((FindUsageProperty)findUsagePropertyForContainingType);
+            }
+
+            var containingSymbol = definition.ContainingSymbol;
+
+            // Containing member should only include fields, properties, methods, or events.  Since ContainingSymbol can return other types, use the return value of GetMemberType to restrict to members only.)
```

Comment:
i would prefer if you just checked for that exact set.  it's a bit weird to need to use GetMemberType for this.  For example, i would have to know that GetMemberType has some meaning on a constructor/destructor.